mailbox/backup/src/main/java/org/apache/james/mailbox/backup/InternalDateExtraField.java

```diff
@@ -19,29 +19,29 @@
 
 package org.apache.james.mailbox.backup;
 
+import java.time.Instant;
+import java.time.ZonedDateTime;
 import java.util.Date;
 import java.util.Optional;
 
 import org.apache.commons.compress.archivers.zip.ZipShort;
 
-public class InternalDateExtraField extends LongExtraField {
+public class InternalDateExtraField extends StringExtraField {
 
     public static final ZipShort ID = new ZipShort(0x6F61); // "ao" in little-endian
 
     public InternalDateExtraField() {
         super();
     }
 
-    public InternalDateExtraField(long time) {
-        super(time);
-    }
-
     public InternalDateExtraField(Date date) {
-        super(date.getTime());
+        super(Optional.of(date.toInstant().toString()));
     }
 
     public InternalDateExtraField(Optional<Date> date) {
-        super(date.map(Date::getTime));
+        super(date
+            .map(Date::toInstant)
```

Comment:
By using $Instant$ we loose the information about the timezone.

Furstermore, we should avoid "toString" calls.

I propose to rely on : 

```
DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(zonedDateTime)
ZonedDateTime.parse(dateAsString, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
And how the date looks like: "2011-12-03T10:15:30+01:00"
```

Reply:
do you want to use `Optional<ZonedDateTime>` instead of `Optional<Date>` as an constructor parameter? Since, `message.getInternalDate()` returns a `Date`, then we do not have timezone information. Suppose we use the machine time zone. I pushed updated into `trantienduchn`:`JAMES-2436`